fix: fix pom for artifactory deploy

cprice404 opened the pull request:
After this change, and with appropriate ~/.m2/settings.xml file,
this will deploy successfully to artifactory:

```
sdk use java 11.0.11.hs-adpt
mvn deploy
```

Here's my settings.xml (replace username, and get encrypted password out of "User Profile" page in artifactory):

```
<?xml version="1.0" encoding="UTF-8"?>
<settings xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.2.0 http://maven.apache.org/xsd/settings-1.2.0.xsd" xmlns="http://maven.apache.org/SETTINGS/1.2.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <servers>
    <server>
      <username>YOUR_USERNAME</username>
      <password>YOUR_PASSWORD</password>
      <id>momento-java</id>
    </server>
    <server>
      <username>YOUR_USERNAME</username>
      <password>YOUR_PASSWORD</password>
      <id>momento-java-snapshots</id>
    </server>
  </servers>
  <profiles>
    <profile>
      <repositories>
        <repository>
          <snapshots>
            <enabled>false</enabled>
          </snapshots>
          <id>momento-java</id>
          <name>maven-java-local</name>
          <url>https://momento.jfrog.io/artifactory/maven-java-local</url>
        </repository>
        <repository>
          <snapshots />
          <id>momento-java-snapshots</id>
          <name>maven-java-local</name>
          <url>https://momento.jfrog.io/artifactory/maven-java-local</url>
        </repository>
      </repositories>
      <pluginRepositories>
        <pluginRepository>
          <snapshots>
            <enabled>false</enabled>
          </snapshots>
          <id>momento-java</id>
          <name>maven-java-local</name>
          <url>https://momento.jfrog.io/artifactory/maven-java-local</url>
        </pluginRepository>
        <pluginRepository>
          <snapshots />
          <id>momento-java-snapshots</id>
          <name>maven-java-local</name>
          <url>https://momento.jfrog.io/artifactory/maven-java-local</url>
        </pluginRepository>
      </pluginRepositories>
      <id>artifactory</id>
    </profile>
  </profiles>
  <activeProfiles>
    <activeProfile>artifactory</activeProfile>
  </activeProfiles>
</settings>


```

cprice404 committed:
fix: fix pom for artifactory deploy

After this change, and with appropriate ~/.m2/settings.xml file,
this will deploy successfully to artifactory:

```
sdk use java 11.0.11.hs-adpt
mvn deploy
```

cprice404 commented:
I did one deploy manually so there are some artifacts up there now:

https://momento.jfrog.io/ui/repos/tree/General/maven-java-local/com/google/code/simple-spring-memcached/momento-provider

You guys should test this before you send them the instructions, but they should be able to pull the artifacts by adding this to their consuming poms:

```
        <repository>
            <id>momento-java</id>
            <name>maven-java-public</name>
            <url>https://momento.jfrog.io/artifactory/maven-java-public</url>
        </repository>
```

eaddingtonwhite reviewed (approved):
Nice! So did you end up having make a new repo in artifactory for mvn published projects vs published w/ gradle? is maven-java-local the name of new one? vs maven-local before?

cprice404 commented:
I did make a new repo in artifactory, so I could get it to tell me the dang authentication config :)  Now that I know what that config is supposed to look like though, I should test it and see if it works with the old repo, and if so, delete the new one.  Will ping you on that in a sec.

cprice404 committed:
fix: use original gradle/maven repo so we can get rid of the new one

Had created a second maven repo on artifactory because we couldn't
figure out how to set up creds to publish to the gradle one.  That
is a bad user experience because it means customers have to define
two repositories in their pom (one for this project, one for momento
SDK).

This commit fixes the pom to publish to the original gradle repo,
and also adds a README with example settings file.
